### ISSUE TITLE
Error of foreach and node commands

### ISSUE THREAD

**Issue by psparkmath** [pgf-tikz/pgf#1313]
### Brief outline of the bug

I found that `foreach` command does not work well for `\node`.

I wrote the code `\draw (P1) foreach \n in {2,3}{ -- (P\i) };` to draw lines connecting nodes P1, P2, P3, consecutively. But, this code draw lines connecting P1 and P2 and connecting P1 and P3.

If you change `\node` part to `\coordinate`, then `foreach` command works well.

### Minimal working example (MWE)

```latex
\documentclass{article}
\usepackage{tikz}

\begin{document}
\begin{tikzpicture}
\foreach[count=\i] \n in {1,3,2}{
    \node (P\i) at (\i,\n) {P\i};
%   \coordinate (P\i) at (\i,\n);
%   \node at (P\i) {P\i};
}
%\draw (P1) -- (P2) -- (P3);
\draw (P1) foreach \i in {2,3}{ -- (P\i)}; % <--- HERE!
\end{tikzpicture}
\end{document}

**Comment by jasperhabicht** [comment]
This is actually well-known. See, for example: https://tex.stackexchange.com/a/75152/47927

**Comment by hmenke** [comment]
As documented.

![image](https://github.com/pgf-tikz/pgf/assets/1903556/9da35f97-9f64-4225-ac1d-d0532a7b64ab)

There is an undocumented option `foreach[scope iterations=false]` but it's undocumented for a reason, namely that in most of the cases it's broken.

**Comment by muzimuzhi** [comment]
Would be fixed by #1304 too?

**Comment by hmenke** [comment]
Yes. It's also a duplicate of https://github.com/pgf-tikz/pgf/issues/356 and https://github.com/pgf-tikz/pgf/issues/1303 and closely related to https://github.com/pgf-tikz/pgf/issues/1047.
